src/Ocelot.Provider.Kubernetes/Kube.cs

```diff
@@ -35,17 +33,14 @@ public virtual async Task<List<Service>> GetAsync()
             .ResourceClient(client => new EndPointClientV1(client))
             .GetAsync(_configuration.KeyOfServiceInK8s, _configuration.KubeNamespace);
 
-        _services.Clear();
-        if (endpoint?.Subsets.Count != 0)
-        {
-            _services.AddRange(BuildServices(_configuration, endpoint));
-        }
-        else
+        if ((endpoint?.Subsets?.Count ?? 0) == 0)
         {
             _logger.LogWarning(() => $"K8s Namespace:{_configuration.KubeNamespace}, Service:{_configuration.KeyOfServiceInK8s}; Unable to use: it is invalid. Address must contain host only e.g. localhost and port must be greater than 0!");
+            return new(0);
```

Comment:
```diff
! This is a generic failed response processing.
```
Therefore, we should revert to the old collection if the response fails, i.e., `Count==0`.
However, "caching" the previous state necessitates synchronization logic.

Reply:
As part of bug fixing, I wouldn't add new logic like caching, as it wasn't here before. At the very least, the cache should not be just storing the previous state. For example, if I downscale the services to 0 pods, I expect that while the pods are terminating (especially when there is some logic on terminate), nobody can call this pod using cached host and port of pod.

But the idea of using an expiring cache would positively impact the `PollKube` logic because if, for some reason, we get 0 services, we will encounter many errors during the polling interval.

I guess it should be a new feature, during the planning of which various pros and cons will be considered.

Reply:
You are correct; it appears to be a new feature.

> For example, if I downscale the services to 0 pods, I expect that while the pods are terminating (especially when there is some logic on terminate), nobody can call this pod using cached host and port of pod.

Although I am not acquainted with scenarios involving pod termination, your point is significant in resolving this debate.
If the Kubernetes client can return zero service instances while a pod is terminating, then receiving an empty collection should not pose a problem.

Reply:
Yes, I was wrong.  I've checked the API and everything should be okay  if there are no pods.

Reply:
What do you suggest here? I can't get you...

Reply:
Agreed! We'll keep it as it is. No problem at all.